sdk/core/Azure.Core/src/Shared/GZipUtf8JsonRequestContent.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.IO;
+using System.IO.Compression;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal class GZipUtf8JsonRequestContent: RequestContent
+    {
+#pragma warning disable CA2213 // Disposable fields should be disposed
+        private GZipStream _gzip;
+        private readonly MemoryStream _stream;
+#pragma warning restore CA2213 // Disposable fields should be disposed
+        private readonly RequestContent _content;
+
+        public Utf8JsonWriter JsonWriter { get; }
+
+        public GZipUtf8JsonRequestContent(RequestContent content)
+        {
+            _stream = new MemoryStream();
+            _gzip = new GZipStream(_stream, CompressionMode.Compress, true);
+            _content = Create(_stream);
+            JsonWriter = new Utf8JsonWriter(_gzip);
+            content.WriteTo(_gzip, default);
+            Flush();
+        }
+
+        public GZipUtf8JsonRequestContent()
+        {
+            _stream = new MemoryStream();
+            _gzip = new GZipStream(_stream, CompressionMode.Compress, true);
+            _content = Create(_stream);
+            JsonWriter = new Utf8JsonWriter(_gzip);
+        }
+
+        public override async Task WriteToAsync(Stream stream, CancellationToken cancellation)
+        {
+            await FlushAsync().ConfigureAwait(false);
+            await _content.WriteToAsync(stream, cancellation).ConfigureAwait(false);
+        }
+
+        public override void WriteTo(Stream stream, CancellationToken cancellation)
+        {
+#if NETFRAMEWORK
```

Comment:
Why is this needed?

Reply:
I think this because in .NET framework, we have to flush before we dispose the gzip stream. @christothes

Reply:
yeah - this explains it https://github.com/dotnet/runtime/issues/15371

Reply:
Interesting! Would it be worth linking to the issue in a comment?